lib/src/Uncertainty/Bayesian/Gibbs.cxx

```diff
@@ -96,28 +143,35 @@ Point Gibbs::getRealization() const
   // see which components will need to recompute the posterior
   if (samplesNumber_ == 0)
   {
-    Point lp(getDimension());
-    for (UnsignedInteger j = 0; j < samplers_.getSize(); ++ j)
+    Point lp(nbSamplers);
+    for (UnsignedInteger j = 0; j < nbSamplers; ++ j)
       lp[j] = samplers_[j].computeLogPosterior(samplers_[j].getImplementation()->currentState_);
-    recomputeLogPosterior_ = Indices(getDimension());
-    for (UnsignedInteger j = 0; j < samplers_.getSize(); ++ j)
-      recomputeLogPosterior_[j] = (lp[j] != lp[(j + getDimension() - 1) % getDimension()]) ? 1 : 0;
-  }
+    recomputeLogPosterior_ = Indices(nbSamplers);
+    for (UnsignedInteger j = 0; j < nbSamplers; ++ j)
+      recomputeLogPosterior_[j] = (lp[j] != lp[(j + nbSamplers - 1) % nbSamplers]) ? 1 : 0;
```

Comment:
Nice candidate to the IOCCC competition ;-)

Reply:
OK, I removed the modulo operation, which costs one whole additional line!